Yearn-agents/InstantWithdrawl/src/agent.ts

```diff
@@ -0,0 +1,71 @@
+import { BlockEvent, Finding, HandleBlock, getJsonRpcUrl } from "forta-agent";
+import axios, { AxiosPromise } from "axios";
+import { generateFinding, getAccounts, Mapping } from "./utils";
+import Web3 from "web3";
+import runServer from "./blockchain";
+import { withdrawAbi } from "./abi";
+import BigNumber from "bignumber.js";
+
+const providerHandleBlock = (Web3: any, axios: any): HandleBlock => {
+  return async (blockEvent: BlockEvent) => {
+    const findings: Finding[] = [];
+
+    const { mapping, users } = await getAccounts(axios);
+
+    const vaults = Object.keys(mapping);
+
+    const server = await runServer(Array.from(users) as string[]);
+
+    const provider = server.provider;
+
+    const web3 = new Web3(provider);
+
+    const promises: Promise<any>[] = [];
+
+    vaults.forEach(async (vaultAddress: string) => {
+      const vault = mapping[vaultAddress];
+
+      vault.forEach(async (obj, index) => {
+        const userAddress = obj.account;
+        const balance = obj.balance;
+
+        const contract = new web3.eth.Contract(
+          withdrawAbi as any,
+          vaultAddress
+        );
+
+        promises.push(
+          contract.methods
+            .withdraw(balance, userAddress, 0)
+            .send({ from: userAddress })
+            .then(() => {
+              contract.methods
+                .balanceOf(userAddress)
+                .call({}, blockEvent.blockNumber)
+                .then((res: string) => {
+                  if (!(res === balance)) {
```

Comment:
Are you sure about that condition? I think it should check if the balance after withdrawing (`res`) is bigger than zero or something like that